Update release note with OpenAPI changes (#2185)
ballerina-platform/ballerina-release: release-notes/swan-lake-release-notes/swan-lake-2201.3.0-release-note.md

- The generated executable contains the modules in the current package, their dependencies, Ballerina runtime, and statically linked native code from the JDK.
- `bal build --native`

#### OpenAPI Tool
- Added support to allow user to add an openAPI specification to generate the IDL client as follows.
```ballerina
client "./openapi.yaml" as foo; 
```
- Introduced `@openapi:ClientConfig` annotation to add additional details to create IDL client.
```ballerina

> @openapi:ClientConfig {
>     tags: ["store"],
>     operations: ["op1", "op2"],
>     license: "/path/to/license_file.txt",
>     isResource: true // (default value => true),
>     nullable: false // (default value => false)
> }
> client "./openapi.yaml" as foo;

Reviewer: ```suggestion
@openapi:ClientConfig {
    tags: ["store"],
    operations: ["op1", "op2"],
    license: "/path/to/license_file.txt",
    isResource: true, // default value => true
    nullable: false // default value => false
}
client "./openapi.yaml" as foo;
```

- The generated executable contains the modules in the current package, their dependencies, Ballerina runtime, and statically linked native code from the JDK.
- `bal build --native`

#### OpenAPI Tool

> - Introduced non-Ballerina IDL client declaration support for openAPI specifications. With the support, openAPI specifications can be imported(i.e. declared) at module level and, consumed as follows.

Reviewer 2: ```suggestion
- Introduced non-Ballerina IDL client declaration support to import OpenAPI specifications to declare and consume them at the module level as follows.
```

Reviewer 2: @NipunaRanasinghe - Please check the changes.

Reviewer: This seems incorrect as the client declaration is only getting declared (but not getting consumed) at the module level. Shall we modify accordingly?

Reviewer 2: @lnash94 - Please change as follows:

- Introduced non-Ballerina IDL client declaration support for OpenAPI specifications. With this support, OpenAPI specifications can now be imported (i.e., declared) at the module level and consumed as follows.

```ballerina

> // Add the client import with OpenAPI specification.

Reviewer 2: ```suggestion
// Add the client import with the OpenAPI specification.
```

client "./openapi.yaml" as weather; 

public function main() returns error? {

>     // Initialize the client , which is generated by IDL import.

Reviewer: ```suggestion
    // Initializes the client which is generated from the IDL import.
```

    weather:client weatherClient = check new ();

>     // Consume the `/weather` API, that has been given in the OpenAPI specification.

Reviewer: ```suggestion
    // Consume the `/weather` resource defined in the OpenAPI specification.
```

    weather:CurrentWeatherData weatherResult = check weatherClient->/weather("Colombo");
}
```

> Also introduced `@openapi:ClientConfig` annotation to configure the client generation.

Reviewer 2: ```suggestion
Also, introduced the `@openapi:ClientConfig` annotation to configure the client generation.
```

> - Also introduced `@openapi:ClientConfig` annotation to add additional details to create IDL client.

Reviewer: ```suggestion
Also introduced `@openapi:ClientConfig` annotation to configure the client generation.
```
